From 2396bcb95a6ab89ebc73e39e084ea61a2e8096e3 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Thu, 10 May 2012 10:26:12 +0200
Subject: [PATCH] remoteproc: zynq: Turn on SMP mode when probing failed.

When probing failed, Linux can return to SMP mode again.
Plus correct email address and add module author.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 drivers/remoteproc/zynq_remoteproc.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/remoteproc/zynq_remoteproc.c b/drivers/remoteproc/zynq_remoteproc.c
index b84bc25..d436efe 100644
--- a/drivers/remoteproc/zynq_remoteproc.c
+++ b/drivers/remoteproc/zynq_remoteproc.c
@@ -1,7 +1,7 @@
 /*
  * Zynq Remote Processor driver
  *
- * Copyright (C) 2012 Michal Simek <monstr@monst.eu>
+ * Copyright (C) 2012 Michal Simek <monstr@monstr.eu>
  * Copyright (C) 2012 PetaLogix
  *
  * Based on origin OMAP Remote Processor driver
@@ -286,6 +286,11 @@ ipi_fault:
 irq_fault:
 	clear_irq(pdev);
 
+	/* Cpu can't be power on - for example in nosmp mode */
+	ret = cpu_up(1);
+	if (ret)
+		dev_err(&pdev->dev, "Can't power on cpu1 %d\n", ret);
+
 	return ret;
 }
 
@@ -336,5 +341,6 @@ module_platform_driver(zynq_remoteproc_driver);
 module_param(firmware, charp, 0);
 MODULE_PARM_DESC(firmware, "Override the firmware image name. Default value in DTS.");
 
+MODULE_AUTHOR("Michal Simek <monstr@monstr.eu");
 MODULE_LICENSE("GPL v2");
 MODULE_DESCRIPTION("Zynq remote processor control driver");
-- 
1.7.0.4

